protocol/resource/offering/offering.go

```diff
@@ -108,3 +108,13 @@ func (o Offering) Sign(bearerDID did.BearerDID) error {
 
 	return nil
 }
+
+// Parse parses a json string into a typed Offering
+func (o Offering) Parse(offeringJSON string) (Offering, error) {
```

Comment:
This doesn't seem to need to be a method? I think it can just be a free function

Reply:
sorry, should have opened as draft haha. refactoring this